app/code/Magento/Customer/Controller/Adminhtml/Group/Edit.php

```diff
@@ -1,22 +1,49 @@
 <?php
 /**
- *
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
 namespace Magento\Customer\Controller\Adminhtml\Group;
 
 use Magento\Framework\App\Action\HttpGetActionInterface as HttpGetActionInterface;
 
+/**
+ * Class Edit
+ *
+ * @package Magento\Customer\Controller\Adminhtml\Group
+ */
 class Edit extends \Magento\Customer\Controller\Adminhtml\Group implements HttpGetActionInterface
 {
     /**
-     * Edit customer group action. Forward to new action.
+     * Edit customer group.
      *
-     * @return \Magento\Backend\Model\View\Result\Forward
+     * @return \Magento\Framework\Controller\ResultInterface
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
      */
     public function execute()
     {
-        return $this->resultForwardFactory->create()->forward('new');
+        $groupId = $this->getRequest()->getParam('id');
+
+        if (isset($groupId)) {
+            $group = $this->groupRepository->getById($groupId);
+            if (!$group->getCode()) {
+                $this->messageManager->addErrorMessage(__('This Group no longer exists.'));
+                /** @var \Magento\Backend\Model\View\Result\Redirect $resultRedirect */
+                $resultRedirect = $this->resultRedirectFactory->create();
+                return $resultRedirect->setPath('*/*/');
+            }
+        }
+        $this->_coreRegistry->register('customer_group', $group);
```

Comment:
`$group` will not be defined if `$groupId` is not set. I.e. for new customer group

Reply:
thanks I will take a look at it

Reply:
@sivaschenko the requested change has been made.

Reply:
@lewisvoncken thanks for the change, can you please also check the travis build fail

Reply:
@sivaschenko is this blocking for the first version?